lib/cylc/param_expand.py

```diff
@@ -64,15 +64,15 @@ def expand(template, params, results, values=None):
 from parsec.OrderedDict import OrderedDictWithDefaults
 
 # To split runtime heading name lists.
-REC_NAMES = re.compile(r'(?:[^,<]|\<[^>]*\>)+')
+REC_NAMES = re.compile(r'(?:[^,<]|<[^>]*>)+')
```

Comment:
I think this one may be dangerous:

```console
$ sed -r 's/<foo>/x/' <<<'a<foo>b
axb
$ sed -r 's/\<foo\>/x/' <<<'a<foo>b'
a<x>b
```

Reply:
Can't remember if Python's `re` support `\<` and `\>` for word boundaries or not. I normally use `\b` for word boundary.

Reply:
Ditto.

Reply:
```console
$ python -c "import re; print(re.sub(r'\<fun\>', r'a fun b', 'x'))"
x
```

Reply:
```
re.sub(pattern, repl, string, count=0, flags=0)
```
Args 'repl' and 'string' switched above? 

```
python -c "import re; print(re.sub(r'\<fun\>', 'x', r'a fun b'))"
a fun b

python -c "import re; print(re.sub(r'\bfun\b', 'x', r'a fun b'))"
a x b
```

Reply:
Anyhow,

> I think this one may be dangerous:

Not dangerous, right?

Reply:
> Can't remember if Python's re support \< and \> for word boundaries or not

Apparently not!

```python
import re

FOO = '''
    foo => <bar> => baz
    foo<bar> => baz
    foo => baz<bar>
'''

print('before')
REC_NAMES = re.compile(r'\<[^>]*\>')
for match in REC_NAMES.finditer(FOO):
    print(match)


print('\nafter')
REC_NAMES = re.compile(r'<[^>]*>')
for match in REC_NAMES.finditer(FOO):
    print(match)

print('\nword break')
REC_NAMES = re.compile(r'\b[^>]*\b')
for match in REC_NAMES.finditer(FOO):
    print(match)
```

```
before
<re.Match object; span=(12, 17), match='<bar>'>
<re.Match object; span=(32, 37), match='<bar>'>
<re.Match object; span=(59, 64), match='<bar>'>

after
<re.Match object; span=(12, 17), match='<bar>'>
<re.Match object; span=(32, 37), match='<bar>'>
<re.Match object; span=(59, 64), match='<bar>'>

word break
<re.Match object; span=(5, 8), match='foo'>
<re.Match object; span=(8, 8), match=''>
<re.Match object; span=(13, 16), match='bar'>
<re.Match object; span=(16, 16), match=''>
<re.Match object; span=(21, 36), match='baz\n    foo<bar'>
<re.Match object; span=(36, 36), match=''>
<re.Match object; span=(41, 52), match='baz\n    foo'>
<re.Match object; span=(52, 52), match=''>
<re.Match object; span=(56, 63), match='baz<bar'>
<re.Match object; span=(63, 63), match=''>

```

> Not dangerous, right?

Seems this is safe in Python.